Jenkinsfile

```diff
@@ -70,6 +70,20 @@ pipeline {
                 sh 'find . -name "hs_err_*" -exec echo {} \\; -exec grep "^# " {} \\;'
             }
         }
+
+        stage('Deploy') {
+            when {
+                allOf {
+                    branch 'master'
+                    equals expected: 0, actual: currentBuild.rawBuild.getAction(AbstractTestResultAction.class).failCount
```

Comment:
I think this is way too specific, the build could fail before it tries to run any test.
Why not use a `failure` block like https://github.com/infinispan/infinispan/pull/5897/files?

Reply:
I agree. @rigazilla did a pretty nice job by extracting `failure` and `success` scenarios in https://github.com/infinispan/infinispan/pull/5897. I guess I should put this there...